### ISSUE TITLE
Hydra and a Mikrotic router...

### ISSUE THREAD

**Issue by JackZaitsev** [vanhauser-thc/thc-hydra#104]
Hello, thanks for the nice software.

I'm having trouble making a attack in my Microtik router, I run a large network for distributing Internet access in my town (its a small town and broadband is not available for everyone), so I'm testing its vulnerabilities against many types of attacks, but I cant find a way to make a successful attack with hydra (which is a good thing), just want to make sure it is not possible and not something I'm missing, the details will follow.

Host=192.168.123.10:9090
User-Agent=Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0
Accept=text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language=en-US,en;q=0.5
Accept-Encoding=gzip, deflate
Referer=http://192.168.123.10:9090/webfig/
Content-Length=117
Content-Type=text/plain;charset=UTF-8
Connection=keep-alive
POSTDATA=ÄÄÄjÄÄÄÄ&&Ã¥-0Ã¢O/Ã½Â¾wÃ_ÃvÄÄ!@#$%^&*()_+:3|~ÄÄÄÄÄÄÄÄÃÃ´odÂ½Y;Â.ÃÂ¯aDÂÂCÃÂPZÃ**THISLOGIN**

"THISLOGIN" being the login I input on the login field in the page form, I have no idea of what those weird characters are, I'm guessing some encoding for the pass, but I could not find a way to decode it.

So I guess my hydra command would look like this:

hydra -l admin -P passfile -V -t 4 -s 9090 192.168.123.10 http-get "/webfig"

I know I'm not passing the failure message, but I could not find it anywhere, it runs like this, but I don't have any idea if its working or not, the right pass is in the file but it never found it.

So the issue here is, I just cant make it work on this router, I tried many different configurations and some of them don't run, some will and return all or some false positives, and some will only return fails even though the right pass is in the file, can you help me in this? thanks very much man.

**Comment by w1r3dh4ck3r** [comment]
Please help? I just need to know if the problem is on my side or is it a limitation of the program, so I can either move on to another software or keep trying.

**Comment by vanhauser-thc** [comment]
hi,

first, what you do is wrong :)
first: it is a POST request so you must use the http-form-post module.
second: the location for hydra to post to must be the same the browser is posting to. likely /webfig isnt it, but I dont know as you did not post that. 
third: set your browser to NOT use utf8, and redo the login to the the post data in a human readable way.
third: learn how to use the http-form-post module. hydra -U http-form-post   shows you options and some examples, the rest you have to research youself

this is not a help forum. this is solely for reporting bugs.

**Comment by w1r3dh4ck3r** [comment]
Yeeh sorry for asking for help here, I was struggling with this for a week
and exhausted any other sources for help, I tried using the http-form-post
module, but  still get errors, watching the data flow with wireshark I see
that using http-get, there is a request going out (knew it was not going to
work) and with http-form-post, nothing goes out, at least wireshark does
not register it.

About the location for hydra to post, if you mean the same location of the
webform (where you input the login info) it is /webfig/index.html, but I
already tried many other iterations and went over the login form looking
for clues, nothing yet.

Setting my browser not to use utf-8 will not help, the data I posted here
was taken from Burpsuit, unless you tell me that setting the browser that
way will alter what burp receives.

Again, sorry for posting here, and thank you for you assistance.

**Comment by JackZaitsev** [comment]
Yeeh sorry for asking for help here, I was struggling with this for a week
and exhausted any other sources for help, I tried using the http-form-post
module, but  still get errors, watching the data flow with wireshark I see
that using http-get, there is a request going out (knew it was not going to
work) and with http-form-post, nothing goes out, at least wireshark does
not register it.

About the location for hydra to post, if you mean the same location of the
webform (where you input the login info) it is /webfig/index.html, but I
already tried many other iterations and went over the login form looking
for clues, nothing yet.

Setting my browser not to use utf-8 will not help, the data I posted here
was taken from Burpsuit, unless you tell me that setting the browser that
way will alter what burp receives.

Again, sorry for posting here, and thank you for you assistance.
